### DISCUSSION TITLE
Has cart global ID format changed recently ?

### DISCUSSION THREAD

**Discussion by Mouarius** [Shopify/storefront-api-feedback#239] [Help] [answered]
Hello everyone,

I am storing the global id of a Cart object ([doc here](https://shopify.dev/docs/api/storefront/2023-10/objects/Cart)) generated with the `useCart` hook ([doc here](https://shopify.dev/docs/api/hydrogen-react/2023-10/hooks/usecart)) of the Hydrogen React lib. I assume that this is not really particular to the context of Hydrogen React, but rather to the Storefront API.

It used to be in this format : `"gid://shopify/Cart/10079785100"` as described in the [documentation](https://shopify.dev/docs/api/storefront/2023-10/scalars/ID), but recently I received an error because it exceeded the size I allowed it to take in my database (100 characters) and changed to something like this, with a `key` property : `gid://shopify/Cart/azerazerazerazerazerazerazerazerazer?key=azerazerazerazerazerazerazer`

I am the only one getting this issue ?
Why such a breaking change even though I'm using a fixed version of the API to `2023-10` ?

Thanks four your answers !

**Comment by corygibbons** [comment, marked as answer]
Changelog entry from earlier this month: https://shopify.dev/changelog/storefront-api-cart-id-now-includes-key-param

Seems like a pretty short grace period, especially when it's applied retroactively 🤷‍♂️

**Reply by Mouarius** [reply]
Thanks a lot, I couldn't find any information about this ! I'll subscribe to the RSS right now ! 🙏🏼

**Reply by vixdug** [reply]
Hey @Mouarius - Always a good idea to subscribe to the changelog. A bit more information and why this change is not tied to a version is that this key ensures that whoever created the cart is the one interacting with the cart. A Cart ID is a public identifier - you can share it, and you can see it in other places of the buyer journey. 

If you exclude the key, your buyer is able to still continue the checkout process, however any personal information such as their email will be removed incase it is a different user.
